contracts/modules/TransactionManager.sol

```diff
@@ -153,6 +157,51 @@ abstract contract TransactionManager is BaseModule {
         return whitelistAfter > 0 && whitelistAfter < block.timestamp;
     }
 
+    /**
+    * @notice Enable the static calls required to make the wallet compatible with the ERC1155TokenReceiver 
+    * interface (see https://eips.ethereum.org/EIPS/eip-1155#erc-1155-token-receiver). This method only 
+    * needs to be called for wallets deployed in version lower or equal to 2.4.0 as the ERC1155 static calls
+    * are not available by default for these versions of BaseWallet
+    * @param _wallet The target wallet.
+    */
+    function enableERC1155TokenReceiver(
+        address _wallet
+    )   
+        external
+        onlySelf()
+    {
+        IWallet(_wallet).enableStaticCall(address(this), ERC165_INTERFACE);
+        IWallet(_wallet).enableStaticCall(address(this), ERC1155_RECEIVED);
+        IWallet(_wallet).enableStaticCall(address(this), ERC1155_BATCH_RECEIVED);
+        emit ERC1155TokenReceiverEnabled(_wallet);
+    }
+
+    /**
+     * @notice Returns true if this contract implements the interface defined by
+     * `interfaceId` (see https://eips.ethereum.org/EIPS/eip-165).
+     */
+    function supportsInterface(bytes4 interfaceID) external view returns (bool) {
```

Comment:
Maybe move this method in the Callback section with `isValidSignature` and the `fallback` ?